api/v3/utils.php

```diff
@@ -1395,7 +1395,7 @@ function _civicrm_api3_basic_create_fallback($bao_name, $params) {
 function _civicrm_api3_basic_delete($bao_name, &$params) {
   civicrm_api3_verify_mandatory($params, NULL, ['id']);
   _civicrm_api3_check_edit_permissions($bao_name, ['id' => $params['id']]);
-  if (method_exists($bao_name, 'del')) {
+  if (method_exists($bao_name, 'del') && !\Civi\Api4\Utils\ReflectionUtils::isMethodDeprecated($bao_name, 'del')) {
```

Comment:
Won't this alter other v3 api delete calls - in possibly unpredictable ways?

Reply:
(also - can we use a `use` )

Reply:
Yes, but I like to think that it will do so in *predictable* ways. :grin: 

As @aydun and I have been systematically deprecating `BAO::del()` functions, we've done so in a way that makes them functionally identical to the base method `deleteRecord()`, so switching the API from one to the other should have no impact. And APIv4 has already made that switch. Let's see what the tests think about my prediction...

Reply:
OK - I'll let the tests decide

Reply:
Re a "use" statement. That makes me a little nervous (maybe unjustifiably so) because `api/v3/utils.php` is not a class and is not in a namespace and gets included from random places that might crash if the `require_once "api/v3/utils.php"` gets called before our classloader is active.

Reply:
hmm ok